javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -565,6 +608,7 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
         } else if (isEmpty()) {
             return ofAll(elements);
         } else {
+            // TODO should be done via merging of two Vectors, there are too many unnecessary array copies done this way
```

Comment:
If it's merged, I will, but I consider it WIP until I optimize all the usecases I TODOed :)